compiler/rustc_interface/src/queries.rs

```diff
@@ -311,6 +318,12 @@ impl Compiler {
     {
         // Must declare `_timer` first so that it is dropped after `queries`.
         let mut _timer = None;
+        let _guard = rustc_data_structures::defer(|| {
```

Comment:
I think it would make more sense for this to be declared after `queries` so that we store `null` before the destructor of `GlobalCtxt` gains unique access, even though it offers no synchronization.